common/src/main/java/com/wynntils/features/user/ContainerSearchFeature.java

```diff
@@ -0,0 +1,228 @@
+
+        String name = ComponentUtils.getCoded(abstractContainerScreen
+                .getMenu()
+                .getItems()
+                .get(currentContainerType.getNextItemSlot())
+                .getHoverName());
+
+        if (!currentContainerType.nextItemPattern.matcher(name).matches()) {
+            autoSearching = false;
+            return;
+        }
+
+        ContainerUtils.clickOnSlot(
+                currentContainerType.getNextItemSlot(),
+                abstractContainerScreen.getMenu().containerId,
+                GLFW.GLFW_MOUSE_BUTTON_LEFT,
+                abstractContainerScreen.getMenu().getItems());
+    }
+
+    private ContainerType getCurrentContainerType(String title) {
+        if (filterInBank && ContainerType.BANK.getTitlePattern().matcher(title).matches()) {
+            return ContainerType.BANK;
+        }
+
+        if (filterInGuildBank
+                && ContainerType.GUILD_BANK.getTitlePattern().matcher(title).matches()) {
+            return ContainerType.GUILD_BANK;
+        }
+
+        if (filterInGuildMemberList
+                && ContainerType.MEMBER_LIST.getTitlePattern().matcher(title).matches()) {
+            return ContainerType.MEMBER_LIST;
+        }
+
+        return null;
+    }
+
+    private void addSearchWidget(AbstractContainerScreen<?> screen, int renderX, int renderY) {
+        SearchWidget searchWidget = new SearchWidget(
+                renderX + screen.imageWidth - 100, renderY - 20, 100, 20, s -> matchItems(s, screen), (TextboxScreen)
+                        screen);
+
+        if (lastSearchWidget != null) {
+            searchWidget.setTextBoxInput(lastSearchWidget.getTextBoxInput());
+        }
+
+        lastSearchWidget = searchWidget;
+
+        screen.addRenderableWidget(lastSearchWidget);
+    }
+
+    private void matchItems(String search, AbstractContainerScreen<?> screen) {
+        search = search.toLowerCase(Locale.ROOT);
+
+        NonNullList<ItemStack> playerItems = McUtils.inventory().items;
+        for (ItemStack item : screen.getMenu().getItems()) {
+            if (!(item instanceof WynnItemStack wynnItemStack)) continue;
+            if (playerItems.contains(item)) continue;
+
+            String name = ComponentUtils.getUnformatted(item.getHoverName()).toLowerCase(Locale.ROOT);
+
+            boolean filtered = !search.equals("") && name.contains(search) && item.getItem() != Items.AIR;
+
+            wynnItemStack.getProperty(ItemProperty.SEARCH_OVERLAY).setSearched(filtered);
+
+            if (filtered) {
+                autoSearching = false;
+            }
+        }
+    }
+
+    private void forceUpdateSearch() {
+        Screen screen = McUtils.mc().screen;
+        if (lastSearchWidget != null && screen instanceof AbstractContainerScreen<?> abstractContainerScreen) {
+            matchItems(lastSearchWidget.getTextBoxInput(), abstractContainerScreen);
+        }
+    }
+
+    public enum ContainerType {
```

Comment:
This seems to general to be put in a Feature. Move to model or a util class.

Reply:
I moved the enum to not be an inner class, but imo the feature itself can keep the other code, since it is generalized enough.

Reply:
(Even the enum feels like generalized enough since I renamed it to SearchableContainerType)